addons/mail/controllers/search.py

```diff
@@ -0,0 +1,45 @@
+from odoo import http
+from odoo.http import request
+from odoo.osv import expression
+
+
+class MailSearchController(http.Controller):
+    @http.route('/mail/thread/search', methods=['POST'], type='json', auth='user')
+    def mail_thread_search(self, search_term, thread_model, thread_id, **kwargs):
+        domain = []
+        domain = expression.AND([
+            expression.OR([
+                [('body', 'ilike', search_term)],
+                [('tracking_value_ids.field_desc', 'ilike', search_term)],
```

Comment:
Independently from the usage of searching on tracking values, you have to respect field groups, otherwise you are leaking potential information people should not access.

Reply:
I have no idea what you are talking about. Could you elaborate ?